cli/log/flag.go

```diff
@@ -0,0 +1,156 @@
+	OutputStderr = "stderr" // Standard error destination, suitable for error conditions
+)
+
+// RegisterLoggingFlags registers the logging-related flags with the provided cobra command.
+// It adds flags for FormatFlagName, LevelFlagName, and OutputFlagName.
+// The flags are added as persistent flags, meaning they will be available to the command
+// and all its subcommands.
+//
+// Usage examples:
+//
+//	--logformat json     # Output logs in JSON format for machine processing
+//	--logformat text     # Output logs in human-readable text format
+//	--loglevel debug     # Show all logs including debug information
+//	--loglevel info      # Show informational messages and above
+//	--loglevel warn      # Show warnings and errors only (default)
+//	--loglevel error     # Show errors only
+//	--logoutput stdout   # Write logs to standard output
+//	--logoutput stderr   # Write logs to standard error
+//
+// These flags can be combined, for example:
+//
+//	--logformat json --loglevel debug --logoutput stderr
+func RegisterLoggingFlags(flagset *pflag.FlagSet) {
+	enum.Var(flagset, FormatFlagName, []string{
+		FormatJSON,
+		FormatText,
+	}, `set the log output format that is used to print individual logs
+   json: Output logs in JSON format, suitable for machine processing
+   text: Output logs in human-readable text format, suitable for console output`)
+
+	enum.Var(flagset, LevelFlagName, []string{
+		LevelWarn,
+		LevelDebug,
+		LevelInfo,
+		LevelError,
+	}, `sets the logging level
+   debug: Show all logs including detailed debugging information
+   info:  Show informational messages and above
+   warn:  Show warnings and errors only (default)
+   error: Show errors only`)
+
+	enum.Var(flagset, OutputFlagName, []string{
+		OutputStdout,
+		OutputStderr,
+	}, `set the log output destination
+   stdout: Write logs to standard output (default)
+   stderr: Write logs to standard error, useful for separating logs from normal output`)
+}
+
+// GetBaseLogger creates and returns a new slog.Logger instance based on the command's flags.
+// It configures the logger with the specified log level, format, and output destination.
+// Returns an error if any of the flag values are invalid or if there's an issue
+// retrieving the flag values.
+func GetBaseLogger(cmd *cobra.Command) (*slog.Logger, error) {
+	logLevel, err := loggerLevelFromCommand(cmd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get log level: %w", err)
+	}
+
+	format, err := enum.Get(cmd.Flags(), FormatFlagName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the log format from the command flag: %w", err)
+	}
+
+	output, err := enum.Get(cmd.Flags(), OutputFlagName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the log output from the command flag: %w", err)
+	}
+
+	var outputWriter io.Writer
+	switch output {
+	case OutputStdout:
+		outputWriter = cmd.OutOrStdout()
+	case OutputStderr:
+		outputWriter = cmd.OutOrStderr()
+	}
```

Comment:
Do you also need a default here erroring with invalid choice?

Reply:
yes, good catch, thanks!